pkg/api/v1/conversion.go

```diff
@@ -139,17 +140,42 @@ func addFastPathConversionFuncs(scheme *runtime.Scheme) error {
 	return nil
 }
 
+func Convert_v1_Service_To_api_Service(in *Service, out *api.Service, s conversion.Scope) error {
+	// If the user specifies an alpha annotation, move it to beta (#33625).
+	if valueAlpha, ok := in.Annotations[apiservice.AlphaAnnotationHealthCheckNodePort]; ok {
+		in.Annotations[apiservice.BetaAnnotationHealthCheckNodePort] = valueAlpha
+	}
+	if valueAlpha, ok := in.Annotations[apiservice.AlphaAnnotationExternalTraffic]; ok {
+		in.Annotations[apiservice.BetaAnnotationExternalTraffic] = valueAlpha
+	}
+	return autoConvert_v1_Service_To_api_Service(in, out, s)
+}
+
+func Convert_api_Service_To_v1_Service(in *api.Service, out *Service, s conversion.Scope) error {
+	// If the system writes a beta annotation, move it to alpha (#33625).
+	if valueBeta, ok := in.Annotations[apiservice.BetaAnnotationHealthCheckNodePort]; ok {
+		in.Annotations[apiservice.AlphaAnnotationHealthCheckNodePort] = valueBeta
+	}
+	if valueBeta, ok := in.Annotations[apiservice.BetaAnnotationExternalTraffic]; ok {
+		in.Annotations[apiservice.AlphaAnnotationExternalTraffic] = valueBeta
+	}
+
+	return autoConvert_api_Service_To_v1_Service(in, out, s)
```

Comment:
Clarify? 

Going from v1 -> internal type, alpha clobbers beta. This will happen through kubectl etc.

Going from internal type -> v1, beta clobbers, because our internal system components do this (so assignment of healthcheck-nodeport shouldn't get clobbered by emptystring, for example). 

We store in etcd as v1. On bootup apiserver reads and converts.


> User reads it back in newer client and changes the beta config, writes back

So this is going from v1->internal? that will just overwrite beta with alpha. 
If we end up with 2 versions, alpha wins, because presumably alpha is holding some resources from before the upgrade.